### PR TITLE
fix many CI related issues

### PR DESCRIPTION
**Pull request by zapashcanon**

### CODE REVIEW

**Comment by filipeom** [comment]
I think because `--with-test` installs alt-ergo it will not compile smtml.0.3.1 since it needs a development version of alt-ergo. 

Maybe we can add this pin in owi to fix the CI:

```
 ["alt-ergo-lib.2.6.0" "git+https://github.com/OCamlPro/alt-ergo.git#7ee96296a5de9588a4d585b3b8abec35f4755425"]
```

**Comment by zapashcanon** [comment]
Thanks. :) I'll try this but shouldn't our dependency on smtml.0.3.1 force opam to use the development version of alt-ergo ?

**Comment by filipeom** [comment]
> Thanks. :) I'll try this but shouldn't our dependency on smtml.0.3.1 force opam to use the development version of alt-ergo ?

I had to remove the pinned dependencies from smtml's .opam file. I only noticed this when I made the last PR to update Owi. But I'll update smtml's .opam file in the opam repository once a new version of alt-ergo is released

**Comment by zapashcanon** [comment]
Ah yes sure, makes sense, they won't let you commit a pin-depends in opam-repo. I'll merge this as is for now. We can remove it later once new alter-ergo and smtml versions are released

File: owi.opam.template [hunk 1]
```diff
@@ -5,4 +5,5 @@ depexts: [
 ]
 pin-depends: [
   [ "crowbar.dev" "git+https://github.com/stedolan/crowbar#1ab53fb088d56578b48301bc4cfb859331a10d78"]
+  ["alt-ergo-lib.2.6.0" "git+https://github.com/OCamlPro/alt-ergo.git#7ee96296a5de9588a4d585b3b8abec35f4755425"]
```
Comment (filipeom) [inline owi.opam.template:8]:
```suggestion
  ["alt-ergo-lib.2.6.0" "git+https://github.com/OCamlPro/alt-ergo.git#7ee96296a5de9588a4d585b3b8abec35f4755425"]
  ["alt-ergo.2.6.0" "git+https://github.com/OCamlPro/alt-ergo.git#7ee96296a5de9588a4d585b3b8abec35f4755425"]
```

File: owi.opam [hunk 2]
```diff
@@ -76,4 +76,5 @@ depexts: [
 ]
 pin-depends: [
   [ "crowbar.dev" "git+https://github.com/stedolan/crowbar#1ab53fb088d56578b48301bc4cfb859331a10d78"]
+  ["alt-ergo-lib.2.6.0" "git+https://github.com/OCamlPro/alt-ergo.git#7ee96296a5de9588a4d585b3b8abec35f4755425"]
```
Comment (filipeom) [inline owi.opam:79]:
```suggestion
  ["alt-ergo-lib.2.6.0" "git+https://github.com/OCamlPro/alt-ergo.git#7ee96296a5de9588a4d585b3b8abec35f4755425"]
  ["alt-ergo.2.6.0" "git+https://github.com/OCamlPro/alt-ergo.git#7ee96296a5de9588a4d585b3b8abec35f4755425"]
```

**Comment by filipeom** [comment]
It was missing the alt-ergo solver pin as well I believe

**Comment by zapashcanon** [comment]
Done. Let's see if it works now...

I opened an issue (#455) so we don't forget to investigate why the backtraces are turned on.

**Comment by zapashcanon** [comment]
OK, looks like we are hitting https://github.com/ocaml/setup-ocaml/issues/897 on macOS.

For the linux build, I'm not sure what's going on, there's no proper error displayed...

**Comment by filipeom** [comment]
> For the linux build, I'm not sure what's going on, there's no proper error displayed...

Looks like we're running out of diskspace in the runner. You can see the errors at the top of the CI result

image

**Comment by zapashcanon** [comment]
Aww, thanks, I missed it somehow... Let's see now!

**Pull request merged**